File: model/src/main/java/com/SReMake/model/system/Resources.java
package com.SReMake.model.system;

import com.SReMake.model.BaseEntity;
import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.OneToOne;
import org.babyfish.jimmer.sql.Table;
import org.jetbrains.annotations.Nullable;

@Entity
@Table(name = "srm_resources")
public interface Resources extends BaseEntity {
    String resources();

    String action();

    Type type();

    @Nullable
    @OneToOne
    Resources parent();

    public enum Type {
        ROUTER,
        BUTTON,
        MENU
    }

}

File: model/src/main/kotlin/com/SReMake/model/system/Resources.kt
package com.SReMake.model.system

import com.SReMake.model.BaseEntity
import org.babyfish.jimmer.sql.*

@Entity
@Table(name = "srm_resources")
interface Resources : BaseEntity {
    val resources: String
    val action: String?
    val type: Type

    @ManyToOne
    @JoinColumn(foreignKeyType = ForeignKeyType.FAKE)
    val parent: Resources?


}

enum class Type {
    ROUTER,
    BUTTON,
    MENU
}